pkg/operators/oci-handler/oci.go

```diff
@@ -206,6 +206,7 @@ func (o *OciHandlerInstance) init(gadgetCtx operators.GadgetContext) error {
 		AllowedRegistriesOptions: oci.AllowedRegistriesOptions{
 			AllowedRegistries: o.ociParams.Get(allowedRegistries).AsStringSlice(),
 		},
+		Logger: gadgetCtx.Logger(),
```

Comment:
Yeah, I was also wondering how to pass the logger to packages that we don't want to depend on gadgetcontext? Maybe putting it into the context as value?

@flyth any better idea?

```go
type ContextKey string

const (
	LoggerCxtKey ContextKey = "logger"
)
```

```go
		// TODO: gadgetCtx.Context() should already carry the logger as value
		ctx := context.WithValue(gadgetCtx.Context(), gadgetcontext.LoggerCxtKey, gadgetCtx.Logger())

		// Make sure the image is available, either through pulling or by just accessing a local copy
		// TODO: add security constraints (e.g. don't allow pulling - add GlobalParams for that)
		err = oci.EnsureImage(ctx, gadgetCtx.ImageName(),
			imgOpts, o.ociParams.Get(pullParam).AsString())
		if err != nil {
			return fmt.Errorf("ensuring image: %w", err)
		}

		...
```

```go
func ensureImage(ctx context.Context, imageStore oras.Target, image string, imgOpts *ImageOptions, pullPolicy string) error {
	logger, ok := ctx.Value(gadgetcontext.LoggerCxtKey).(logger.Logger)
	if !ok {
		logger = logrus.StandardLogger()
		logger.Debug("could not extract logger from context")
	}

	switch pullPolicy {
	...
```

Reply:
> Yeah, I was also wondering how to pass the logger to packages that we don't want to depend on gadgetcontext? Maybe putting it into the context as value?
> 
> @flyth any better idea?
> 
> ```go
> type ContextKey string
> 
> const (
> 	LoggerCxtKey ContextKey = "logger"
> )
> ```
> 
>  ```go
> 		// TODO: gadgetCtx.Context() should already carry the logger as value
> 		ctx := context.WithValue(gadgetCtx.Context(), gadgetcontext.LoggerCxtKey, gadgetCtx.Logger())
> 
> 		// Make sure the image is available, either through pulling or by just accessing a local copy
> 		// TODO: add security constraints (e.g. don't allow pulling - add GlobalParams for that)
> 		err = oci.EnsureImage(ctx, gadgetCtx.ImageName(),
> 			imgOpts, o.ociParams.Get(pullParam).AsString())
> 		if err != nil {
> 			return fmt.Errorf("ensuring image: %w", err)
> 		}
> 
> 		...
> ```
> 
>  ```go
> func ensureImage(ctx context.Context, imageStore oras.Target, image string, imgOpts *ImageOptions, pullPolicy string) error {
> 	logger, ok := ctx.Value(gadgetcontext.LoggerCxtKey).(logger.Logger)
> 	if !ok {
> 		logger = logrus.StandardLogger()
> 		logger.Debug("could not extract logger from context")
> 	}
> 
> 	switch pullPolicy {
> 	...
> ```

I don't see a reason against doing that. 😄